tests/components/edit_channel_header_modal.test.jsx

```diff
@@ -0,0 +1,139 @@
+                onPatchChannel={emptyFunction}
+            />
+        ).dive({disableLifecycleMethods: true});
+
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    test('submitted', () => {
+        const wrapper = shallowWithIntl(
+            <EditChannelHeaderModal
+                channel={channel}
+                ctrlSend={false}
+                requestStatus={RequestStatus.STARTED}
+                onHide={emptyFunction}
+                onPatchChannel={emptyFunction}
+            />
+        ).dive({disableLifecycleMethods: true});
+
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    test('error with intl message', () => {
+        const serverError = {
+            server_error_id: 'model.channel.is_valid.header.app_error',
+            message: 'some error'
+        };
+
+        const wrapper = shallowWithIntl(
+            <EditChannelHeaderModal
+                channel={channel}
+                ctrlSend={false}
+                requestStatus={RequestStatus.NOT_STARTED}
+                onHide={emptyFunction}
+                onPatchChannel={emptyFunction}
+            />
+        ).dive({disableLifecycleMethods: true});
+
+        wrapper.setProps({
+            channel,
+            serverError,
+            ctrSend: false,
+            requestStatus: RequestStatus.FAILURE,
+            onHide: emptyFunction,
+            onPatchChannel: emptyFunction
+        });
+
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    test('error without intl message', () => {
+        const serverError = {
+            server_error_id: 'fake-server-errror',
+            message: 'some error'
+        };
+
+        const wrapper = shallowWithIntl(
+            <EditChannelHeaderModal
+                channel={channel}
+                ctrlSend={false}
+                requestStatus={RequestStatus.NOT_STARTED}
+                onHide={emptyFunction}
+                onPatchChannel={emptyFunction}
+            />
+        ).dive({disableLifecycleMethods: true});
+
+        wrapper.setProps({
+            channel,
+            serverError,
+            ctrSend: false,
+            requestStatus: RequestStatus.FAILURE,
+            onHide: emptyFunction,
+            onPatchChannel: emptyFunction
+        });
+
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    test('patch on save button click', () => {
+        const patchChannel = jest.fn();
+        const wrapper = shallowWithIntl(
+            <EditChannelHeaderModal
+                channel={channel}
+                ctrlSend={false}
+                requestStatus={RequestStatus.NOT_STARTED}
+                onHide={emptyFunction}
+                onPatchChannel={patchChannel}
+            />
+        ).dive({disableLifecycleMethods: true});
+
+        wrapper.find('.save-button').simulate('click');
+
+        expect(patchChannel.mock.calls.length).toBe(1);
+        expect(patchChannel.mock.calls[0][0]).toBe('fake-id', {header: 'Fake Channel'});
+    });
+});
```

Comment:
Test coverage for this component is a bit low. If possible, please add more tests to bring it up. Please see this [test](https://github.com/saturninoabril/mattermost-webapp/blob/3272562cf4cee3092ce6b5ede4fb4035704d1fd8/tests/components/post_view/post_info/post_info.test.jsx) if similar can be applied here.
![screen shot 2017-10-07 at 1 26 24 am](https://user-images.githubusercontent.com/5334504/31290308-954d4e78-aafe-11e7-9fb1-f206eb9f9f46.png)
![screen shot 2017-10-07 at 1 38 54 am](https://user-images.githubusercontent.com/5334504/31290787-54e53830-ab00-11e7-90cc-8a463b1588b9.png)